diff --git a/fabric/v4/model_connection_change_operation.go b/fabric/v4/model_connection_change_operation.go
index ee47126..53971ec 100644
diff --git b/fabric/v4/model_connection_change_operation.go a/fabric/v4/model_connection_change_operation.go
index e9e4c1a..ce72a4d 100644
--- b/fabric/v4/model_connection_change_operation.go
+++ a/fabric/v4/model_connection_change_operation.go
@@ -16,5 +16,5 @@ type ConnectionChangeOperation struct {
 	// path inside document leading to updated parameter
 	Path string `json:"path"`
 	// new value for updated parameter
-	Value *interface{} `json:"value"`
+	Value interface{} `json:"value"`
 }
